modules/candidate_parameters/jsx/candidateInfo.js

```diff
@@ -0,0 +1,326 @@
+                            name ={name}
+                            options ={selectOptions}
+                            value ={value}
+                            onUserInput ={this.setFormData}
+                            ref ={name}
+                            disabled ={disabled}
+                            />
+                        );
+              break;
+            case "dat":
+              extraParameterFields.push(
+                            <DateElement
+                            label ={extraParameters[key2].Description}
+                            name ={name}
+                            value ={value}
+                            onUserInput ={this.setFormData}
+                            ref ={name}
+                            disabled ={disabled}
+                            />
+                        );
+              break;
+            default:
+              extraParameterFields.push(
+                            <TextareaElement
+                            label ={extraParameters[key2].Description}
+                            name ={name}
+                            value ={value}
+                            onUserInput ={this.setFormData}
+                            ref ={name}
+                            disabled ={disabled}
+                            />
+                        );
+          }
+        }
+      }
+
+      var alertMessage = "";
+      var alertClass = "alert text-center hide";
+      if (this.state.updateResult) {
+        if (this.state.updateResult === "success") {
+          alertClass = "alert alert-success text-center";
+          alertMessage = "Update Successful!";
+        } else if (this.state.updateResult === "error") {
+          var errorMessage = this.state.errorMessage;
+          alertClass = "alert alert-danger text-center";
+          alertMessage = errorMessage ? errorMessage : "Failed to update!";
+        }
+      }
+      return (
+            <div>
+                <div className ={alertClass} role="alert" ref="alert-message">
+                    {alertMessage}
+                </div>
+                <FormElement
+                    name ="candidateInfo"
+                    onSubmit ={this.handleSubmit}
+                    ref ="form"
+                    class ="col-md-6"
+                >
+                    <StaticElement
+                        label ="PSCID"
+                        text ={this.state.Data.pscid}
+                    />
+                    <StaticElement
+                        label ="DCCID"
+                        text ={this.state.Data.candID}
+                    />
+                    <SelectElement
+                        label ="Caveat Emptor Flag for Candidate"
+                        name ="flagged_caveatemptor"
+                        options ={this.state.caveatOptions}
+                        value ={this.state.Data.flagged_caveatemptor}
+                        onUserInput ={this.setFormData}
```

Comment:
You should probably reset unused fields, before hiding them on option change. Same with error messages.
Otherwise, you might end up with stuff like this:

Field supposed to only be required for true
![image](https://cloud.githubusercontent.com/assets/6627543/18886754/44402d90-84bf-11e6-964e-33857de6517d.png)

Field should not have a value for false
![image](https://cloud.githubusercontent.com/assets/6627543/18886794/71afcf06-84bf-11e6-9883-4837a5814f6d.png)